infra/Pos/Util/LogSafe.hs

```diff
@@ -52,13 +111,207 @@ logNoticeS  = logMessageS Notice
 logWarningS = logMessageS Warning
 logErrorS   = logMessageS Error
 
--- | Same as 'logMesssage', but log to two loggers, put only insecure
--- version to memmode.
+-- | Same as 'logMesssage', but log to secret logs, put only insecure
+-- version to memmode (to terminal).
 logMessageS
     :: (HasLoggerName m, MonadIO m)
     => Severity
     -> Text
     -> m ()
-logMessageS severity t = execSecureLogWrapped $ do
-    name <- getLoggerName
-    dispatchMessage name severity t
+logMessageS severity t =
+    reify selectSecretLogs $ \s ->
+    execSecureLogWrapped s $ do
+        name <- getLoggerName
+        dispatchMessage name severity t
+
+----------------------------------------------------------------------------
+-- Secure buildables
+----------------------------------------------------------------------------
+
+-- * Secutity log levels
+
+-- Stuff below provides way to use different 'instance Buildable' for writing
+-- to secret and public logs.
+--
+-- Example:
+--
+-- @
+-- logInfoSP $ \sl -> ("Nyan: "%secureF sl build) Nyan
+-- @
+--
+-- this function prints value to secret logs as is, while for public logs it uses
+-- 'instance Buildable (PublicLog Nyan)'.
+--
+-- By analogy, 'logInfoP' will log to public logs only, requiring
+-- 'instance Buildable (PublicLog ...)'. It also provides security level @sl@
+-- parameter which is redundant, but allows to quickly switch
```

Comment:
While reading this paragraph, I'm wondering what you mean by the `sl` parameter being redundant. It'll be evident from the code, I presume, but writing it explicitly in the comments wouldn't hurt either.